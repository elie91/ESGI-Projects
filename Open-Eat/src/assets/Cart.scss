@import "./../App";

.cart {
  position: absolute;
  width: 20rem;
  min-height: 200px;
  top: 75px;
  right: 0;
  border: black;
  background-color: #ffffff;
  z-index: 5;
  box-shadow: 0 .15rem 1.75rem 0 rgba(58,59,69,.15)!important;
  border-radius: .35rem;

  .icon{
    height: 15px;
  }

  .cart-header{
    border-top-left-radius: .35rem;
    border-top-right-radius: .35rem;
    padding: .5rem 1.5rem;
    margin-bottom: 0;
    font-size: .875rem;
    white-space: nowrap;
    background-color: $primary;
    border: 1px solid $primary;
    color: #fff;
    font-weight: bold;
    text-align: center;
  }

  .btn-delete{
    padding: 0 .5rem;
  }

}

.cart-icon {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  padding: 0.5rem 1rem;

  .shopping-icon {
    width: 24px;
    height: 24px;
  }

  .item-count {
    position: absolute;
    font-size: 10px;
    font-weight: bold;
    bottom: 10px;
  }
}

.cursor{
  cursor: pointer;
}

.btn{
  &.btn-icon {
    padding: .5rem;

    &:active, &:hover, &:focus {
      box-shadow: none;
    }
  }
  &.btn-top-left-0{
    border-top-left-radius: 0;
  }
  &.btn-top-right-0{
    border-top-right-radius: 0;
  }
}

@media (max-width: 480px) {

  .cart{
    position: relative;
    width: 100%;
    top: 0;
  }

}
