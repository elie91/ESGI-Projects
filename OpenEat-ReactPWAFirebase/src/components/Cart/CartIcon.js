import React from "react";
import {ReactComponent as ShoppingIcon} from '../../images/shopping-bag.svg';

const CartIcon = ({cartItems, hidden, setCartHidden}) => {

  const count = cartItems.reduce(
    (accumulator, item) => accumulator + item.quantity, 0
  );

  return (
    <div className='cart-icon' onClick={() => setCartHidden(!hidden)}>
      <ShoppingIcon className='shopping-icon'/>
      <span className='item-count'>{count}</span>
    </div>
  );
}

export default React.memo(CartIcon);
