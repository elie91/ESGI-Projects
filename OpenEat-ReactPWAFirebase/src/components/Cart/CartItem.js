import React from 'react';
import { removeItemFromCart } from "../Utils/Utils";
import { ReactComponent as DeleteIcon } from "../../images/trash-solid.svg";

const CartItem = ({item, cartItems, setCartItems}) => (
    <div className='d-flex justify-content-between align-items-center py-2'>
      <div>{item.quantity} x {item.name}</div>
      <button className="btn btn-delete" onClick={() => setCartItems(removeItemFromCart(cartItems, item))}> <DeleteIcon className="icon text-danger"/></button>
    </div>
);

export default React.memo(CartItem);
