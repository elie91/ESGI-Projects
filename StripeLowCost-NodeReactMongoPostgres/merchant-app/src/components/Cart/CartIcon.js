import React from "react";
import {ReactComponent as ShoppingIcon} from '../../images/shopping-bag.svg';
import useProducts from "../../hooks/useProducts";

const CartIcon = () => {
    const {selectors: productsSelectors, actions: productsActions} = useProducts();

    const count = productsSelectors.getCartItems().reduce(
        (accumulator, item) => accumulator + item.quantity, 0
    );
    return (
        <div className='cart-icon' onClick={() => productsActions.toggleCartHidden()}>
            <ShoppingIcon className='shopping-icon'/>
            <span className='item-count'>{count}</span>
        </div>
    );
}

export default React.memo(CartIcon);
