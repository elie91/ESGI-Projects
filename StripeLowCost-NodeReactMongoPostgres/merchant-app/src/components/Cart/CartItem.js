import React from "react";
import useProducts from "../../hooks/useProducts";

const CartItem = ({ item }) => {

  const { actions } = useProducts();

  return (
    <div className="pb-2">
      <div className='d-flex justify-content-between align-items-center'>
        <div>{item.name}</div>
        <div className="d-flex align-items-center">
          <button className="btn btn-icon arrow" onClick={() => actions.removeItemFromCart(item)}>
            -
          </button>
          {item.quantity}
          <button className="btn btn-icon arrow" onClick={() => actions.addItemToCart(item)}>
            +
          </button>
        </div>
        <div>
          {item.price * item.quantity} â‚¬
        </div>
        <div
          className='remove-button'
          onClick={() => actions.clearItemFromCart(item)}> &#10005;</div>
      </div>
    </div>
  );
};

export default CartItem;
