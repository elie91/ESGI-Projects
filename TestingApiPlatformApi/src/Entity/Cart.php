<?php

namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use ApiPlatform\Core\Annotation\ApiSubresource;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ApiResource(
 *     normalizationContext={"groups"={"cart_read"}},
 *     itemOperations={"get","delete"}
 * )
 * @ORM\Entity
 *
 */
class Cart
{
    /**
     * @var int The entity Id
     *
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups({"cart_read"})
     */
    private $id;

    /**
     * @ORM\OneToMany(targetEntity=CartFruit::class, mappedBy="cart", cascade={"persist", "remove"})
     * @ApiSubresource()
     * @Groups({"cart_read"})
     */
    private $cartFruits;

    public function __construct()
    {
        $this->cartFruits = new ArrayCollection();
    }

    /**
     * @return int
     */
    public function getId(): int
    {
        return $this->id;
    }

    /**
     * @param int $id
     */
    public function setId(int $id)
    {
        $this->id = $id;
    }

    /**
     * @return Collection|CartFruit[]
     */
    public function getCartFruits(): Collection
    {
        return $this->cartFruits;
    }

    public function addCartFruit(CartFruit $cartFruit): self
    {
        if (!$this->cartFruits->contains($cartFruit)) {
            $this->cartFruits[] = $cartFruit;
            $cartFruit->setCart($this);
        }

        return $this;
    }

    public function removeCartFruit(CartFruit $cartFruit): self
    {
        if ($this->cartFruits->contains($cartFruit)) {
            $this->cartFruits->removeElement($cartFruit);
            // set the owning side to null (unless already changed)
            if ($cartFruit->getCart() === $this) {
                $cartFruit->setCart(null);
            }
        }

        return $this;
    }
}
