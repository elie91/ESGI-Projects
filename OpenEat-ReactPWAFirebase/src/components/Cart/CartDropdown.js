import React, { useState } from "react";
import CartItem from "./CartItem";
import { createOrder } from "../../firebase";
import { withRouter } from "react-router-dom";

const CartDropdown = ({ cartItems, setCartItems, setCartHidden, user, history }) => {

  const [error, setError] = useState(undefined);
  const [offline, setOffline] = useState(0);

  const _onClick = (e) => {
    e.preventDefault();
    if (cartItems.length > 0){
      if (user) {
        if (!navigator.onLine){
          setOffline(prev => prev + 1);
        }
        if (offline === 0){
          createOrder(user, cartItems).then(function () {
            setCartItems([]);
            setCartHidden(true);
            history.push('/orders');
          });
        }
        if (!navigator.onLine){
          setOffline(prev => prev + 1);
        }
      } else {
        history.push("/login");
        setCartHidden(true);
      }
    }else{
      setError("Vous devez ajouter des produits à votre panier");
    }
  };

  return (
    <div className='cart d-flex justify-content-between flex-column'>
      <div className="cart-header">
        Mon panier
      </div>
      <div className="px-3">
        {
          cartItems.length > 0
            ?
            (cartItems.map((item, index) => <CartItem key={index} item={item} cartItems={cartItems} setCartItems={setCartItems}/>))
            :
            (<p className="px-3 py-2 text-center mb-0">{error ? error : "Votre panier est vide"}</p>)
        }
      </div>
      <div className="text-center">
        <div className="dropdown-divider"/>
        <button className='btn text-primary mb-2' onClick={_onClick}>
          Valider le panier
        </button>
        {offline > 0 && offline < 3 && <div className="text-warning">Votre commande sera envoyé lorsque vous serez à nouveau connecté à Internet</div>}
        {offline > 3 && <div className="text-danger">Calme-toi sur le bouton gros, t'as pas de réseau</div>}
      </div>
    </div>
  );
};

export default withRouter(CartDropdown);
