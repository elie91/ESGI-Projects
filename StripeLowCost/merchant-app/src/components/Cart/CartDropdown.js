import React from "react";
import CartItem from "./CartItem";
import "../../sass/components/_cart-dropdown.scss";
import useProducts from "../../hooks/useProducts";
import {withRouter} from "react-router-dom";

const CartDropdown = ({history}) => {
  const { selectors, actions } = useProducts();

  const _onClick = () => {
    actions.toggleCartHidden();
    history.push('/order')
  }

  return (
    <div className='cart d-flex justify-content-between flex-column'>
      <div className="cart-header">
        Mon panier
      </div>
      <div className="px-3">
        {
          selectors.getCartItems().length > 0
            ?
            (selectors.getCartItems().map((item, index) => <CartItem key={index} item={item}/>))
            :
            (<p className="px-3 py-2 text-center mb-0">Votre panier est vide</p>)
        }
      </div>
      {selectors.getCartItems().length > 0 &&
        <div className="d-flex px-3 justify-content-between">
          <div>
            Total :
          </div>
          <div>
            {selectors.getCartItems().reduce((acc, val) => acc + (val.price * val.quantity), 0)} â‚¬
          </div>
        </div>
      }
      <div className="text-center">
        <div className="dropdown-divider"/>
        <button className='btn text-primary mb-2' onClick={_onClick}>
          Valider le panier
        </button>
      </div>
    </div>
  );
};

export default withRouter(CartDropdown);
