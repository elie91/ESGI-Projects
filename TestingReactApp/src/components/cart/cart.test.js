import React from 'react';
import {shallow, mount} from 'enzyme';

import Cart from './cart.component';
import EditFruit from "../edit-fruit/edit-fruit";

describe('Cart component', () => {

  let wrapper;
  let mockProps;
  let mockCart;
  let mockHandleDelete;

  beforeEach(() => {
    mockCart = [
      {
        '@id': "/fruits/14",
        '@type': "Fruit",
        cartFruits: [],
        id: 14,
        name: "test"
      },
      {
        '@id': "/fruits/15",
        '@type': "Fruit",
        cartFruits: [],
        id: 15,
        name: "toto"
      }
    ]

    mockHandleDelete = jest.fn();

    mockProps = {
      cart: mockCart,
      handleDelete: mockHandleDelete
    };

    wrapper = shallow(<Cart {...mockProps} />)
  });


  it('should render Cart component', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
