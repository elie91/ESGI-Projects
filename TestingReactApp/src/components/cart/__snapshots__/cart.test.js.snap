// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component should render Cart component 1`] = `ShallowWrapper {}`;
